canonicalwebteam/flask_base/app.py

```diff
@@ -20,7 +20,17 @@
 
 
 def set_security_headers(response):
-    response.headers["X-Frame-Options"] = "SAMEORIGIN"
+    add_xframe_options_header = True
+
+    # Check if view_function has exclude_xframe_options_header decorator
+    if flask.request.endpoint in flask.current_app.view_functions:
+        view_func = flask.current_app.view_functions[flask.request.endpoint]
+        add_xframe_options_header = not hasattr(
+            view_func, "_exclude_xframe_options_header"
+        )
+
+    if add_xframe_options_header:
+        response.headers["X-Frame-Options"] = "SAMEORIGIN"
```

Comment:
Could you just add an `if` here to only set the header if it's not already set? That will then allow us to override it with `DENY` or something if needed.